packages/backend/src/routers/oauthRouter.ts

```diff
@@ -0,0 +1,302 @@
+                            req.user.organizationName
+                        }</b></p>
+
+                        <div class="oauth-btn-row">
+                            <button type="submit" name="approve" value="true" class="oauth-btn approve">Approve</button>
+                            <button type="submit" name="approve" value="false" class="oauth-btn deny">Deny</button>
+                        </div>
+                    </form>
+                </div>
+            </body>
+        </html>
+    `);
+});
+
+oauthRouter.post('/authorize', async (req, res, next) => {
+    if (!req.user) {
+        return res.status(401).send('Unauthorized');
+    }
+    if (!req.user.userId || !req.user.organizationUuid) {
+        return res.status(400).send('Missing userUuid or organizationUuid');
+    }
+    if (req.body.approve === 'true') {
+        const oauthService = getOAuthService(req);
+        const oauthReq = new OAuth2Server.Request(req);
+        const oauthRes = new OAuth2Server.Response(res);
+        try {
+            const authorizationCode = await oauthService.authorize(
+                oauthReq,
+                oauthRes,
+                {
+                    userId: req.user.userId,
+                    organizationUuid: req.user.organizationUuid,
+                },
+            );
+            const redirectUri = req.body.redirect_uri;
+            if (!redirectUri) {
+                return res.status(400).json({
+                    error: 'invalid_request',
+                    error_description: 'Missing redirect_uri',
+                });
+            }
+            const url = new URL(redirectUri);
+            url.searchParams.set('code', authorizationCode.authorizationCode);
+            if (req.body.state) {
+                url.searchParams.set('state', req.body.state);
+            }
+            return res.redirect(url.toString());
+        } catch (error) {
+            console.error('Authorization error:', error);
+            const redirectUrl = new URL(req.body.redirect_uri);
+            redirectUrl.searchParams.set('error', 'server_error');
+            if (req.body.state)
+                redirectUrl.searchParams.set('state', req.body.state);
+            return res.redirect(redirectUrl.toString());
+        }
+    } else {
+        // Denied
+        res.set('Content-Type', 'text/html');
+        const redirectUri = req.body.redirect_uri;
+        if (!redirectUri) {
+            return res.status(400).json({
+                error: 'invalid_request',
+                error_description: 'Missing redirect_uri',
+            });
+        }
+        const url = new URL(redirectUri);
+        url.searchParams.set('error', 'access_denied');
+        if (req.body.state) {
+            url.searchParams.set('state', req.body.state);
+        }
+        return res.redirect(url.toString());
+    }
+});
+
+// Post token - use OAuth2Server
+oauthRouter.post('/token', async (req, res, next) => {
```

Comment:
with the authorization token, we use this endpoint to get access token or even refresh token. That logic is all handled in the oauth2server